compiler/rustc_query_system/src/dep_graph/dep_node.rs

```diff
@@ -53,6 +53,19 @@ use std::hash::Hash;
 #[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Encodable, Decodable)]
 pub struct DepNode<K> {
     pub kind: K,
+    // Important - whenever a `DepNode` is constructed, we need to make
```

Comment:
Nit: This comment seems like it applies to the `DepNode` type as a whole and not just the `hash` field so I would move this to above the struct declaration.